FirebaseAuth/Sources/Utilities/FIRAuthRecaptchaVerifier.m

```diff
@@ -0,0 +1,239 @@
+                    action:(FIRAuthRecaptchaAction)action
+                completion:(nullable FIRAuthRecaptchaTokenCallback)completion {
+#if TARGET_OS_IOS
+  [self
+      retrieveRecaptchaConfigForceRefresh:forceRefresh
+                               completion:^(NSError *_Nullable error) {
+                                 if (error) {
+                                   completion(nil, error);
+                                 }
+                                 if (!self.recaptchaClient) {
+                                   NSString *siteKey = @"";
+                                   if ([FIRAuth auth].tenantID == nil) {
+                                     siteKey = self->_agentConfig.siteKey;
+                                   } else {
+                                     siteKey =
+                                         self->_tenantConfigs[[FIRAuth auth].tenantID].siteKey;
+                                   }
+                                   Class RecaptchaClass = NSClassFromString(@"Recaptcha");
+                                   if (RecaptchaClass) {
+                                     SEL selectorWithTimeout = NSSelectorFromString(
+                                         @"getClientWithSiteKey:withTimeout:completion:");
+                                     if ([RecaptchaClass respondsToSelector:selectorWithTimeout]) {
+                                       void (*funcWithTimeout)(
+                                           id, SEL, NSString *, double,
+                                           void (^)(
+                                               id<RCARecaptchaClientProtocol> _Nullable recaptchaClient,
+                                               NSError *_Nullable error)) = (void *)[RecaptchaClass
+                                           methodForSelector:selectorWithTimeout];
+                                       funcWithTimeout(
+                                           RecaptchaClass, selectorWithTimeout, siteKey, 10000.0,
+                                           ^(id<RCARecaptchaClientProtocol> _Nullable recaptchaClient,
+                                             NSError *_Nullable error) {
+                                             if (recaptchaClient) {
+                                               NSLog(@"recaptcha client created.");
+                                               self.recaptchaClient = recaptchaClient;
+                                               [self retrieveRecaptchaTokenWithAction:action
+                                                                           completion:completion];
+                                             } else if (error) {
+                                               NSLog(@"recaptcha client creatation errored.");
+                                               NSLog(@"Error: %@", error);
+                                             }
+                                           });
```

Comment:
Conditionally link to recaptcha sdk with objc reflection, to initialize the recaptcha client.

Reply:
Hey @paulb777 , just integrated with the recaptcha interop pod using reflection. Could you help take a look? Thanks!